app/models/article.rb

```diff
@@ -467,6 +479,7 @@ def detect_human_language
   def async_score_calc
     return if !published? || destroyed?
 
+    Rails.cache.delete(top_comments_cache_name)
```

Comment:
I'm conflicted about the location of this cache busting action, here are my options:

- bust the cache here in `articles` before recalculating article score (which happens when comments change)
- bust the cache in the async action (but I don't know if that's feasible: are sidekiq jobs in a separate dyno? Can they see the Rails cache?)
- bust somewhere in `comments` model since that's what we are really responding to (but: that would require the load of the article object in a comment only action..seems wrong!)

Reply:
The last option seems wrong as you mentioned. I'm probably missing some context, but what is the advantage of busting the cache in the async action vs. in the article model?

Reply:
Yeah: async is generally reserved for operations that could take a while, and/or we don't want to block the user while the action happens. Cache clearing will be very quick, so I don't think doing I asynchronously is advantageous here.